### ISSUE TITLE
508-defect-2 [AXE CORE]: Heading should not be focusable while within the mobile profile menu button

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#23254]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Details

Because the heading has a `tab-index="-1"` and is within a focusable button, an AXE issue is thrown because it is being interpreted as an interactive control.

## Acceptance Criteria

- [ ] An axe-error is no longer thrown for "ensure interactive controls are not nested"

## Steps to Recreate

Run an axe-scan on profile, on mobile, with any logged in user

## Recommended Solution

Make the profile `h4` along with its `tab-index="-1"` the parent, not the child, of the menu button.

## Screenshots or Trace Logs

<img width="1792" alt="Screen Shot 2021-04-14 at 5 06 55 PM" src="https://user-images.githubusercontent.com/14154792/114780181-0a893300-9d45-11eb-8dc0-ca8409b3a62c.png">
<img width="1470" alt="Screen Shot 2021-05-11 at 3 02 58 PM" src="https://user-images.githubusercontent.com/14154792/117871056-f31d6700-b26a-11eb-8260-97a098b70a47.png">

**Comment by ajakabcin** [comment]
Hi @joshkimux! Can this issue be closed now since we implemented the fix documented in #19552?

cc @Samara-Strauss

**Comment by joshkimux** [comment]
Hi @ajakabcin , this issue is still present unfortunately, but can be resolved relatively simply if we're able to move the `h4` and its `tab-index="-1"` outside of the button as the parent. For example:

<img width="1470" alt="Screen Shot 2021-05-11 at 3 02 58 PM" src="https://user-images.githubusercontent.com/14154792/117870469-2ad7df00-b26a-11eb-9039-14d96c1afe23.png">

**Comment by erikphansen** [comment]
I just ran Axe Coconut on staging and am not seeing this violation 🤔

**Comment by joshkimux** [comment]
<img width="1397" alt="Screen Shot 2021-09-08 at 12 44 04 PM" src="https://user-images.githubusercontent.com/14154792/132550425-5c40ed17-b856-497c-aa1f-0fb582a338fd.png">

Just tested again with user 36 and it's still there. Did you run axe on mobile?

**Comment by erikphansen** [comment]
@joshkimux omg yeah it's there now. Sorry. It wasn't showing up in either aXe dev tools earlier :( ¯\_(ツ)_/¯

**Comment by ajakabcin** [comment]
@joshkimux can you take a look for us now to validate? thank you!

**Comment by joshkimux** [comment]
@erikphansen , I took a second look at the code since axe was still catching the error and noticed that the `strong` element has a `tabindex="-1"` which isn't needed since the `h4` is already focusable. 

<img width="1458" alt="Screen Shot 2021-09-08 at 2 16 40 PM" src="https://user-images.githubusercontent.com/14154792/132563078-6111009b-7a83-4a65-83c7-583ff5ac08f3.png">

**Comment by tmitchellgcio** [comment]
Just merged a new PR that should fix this. Tested locally I don't see it anymore

**Comment by joshkimux** [comment]
Confirmed this is fixed! Closing it out. Thanks y'all for the quick work 🥳
